rts/Sim/Misc/Resource.h

```diff
@@ -81,6 +81,7 @@ struct SResourcePack {
 	}
 };
 
+inline SResourcePack operator * (SResourcePack pack, float scale) { return pack *= scale; }
```

Comment:
Shouldn't this be:
`inline SResourcePack operator * (const SResourcePack& pack, float scale) { return pack *= scale; }`

Reply:
No, because that would try ti modify the original `pack` (and wouldn't work with `const &`). We want to return a copy, so if we wanted to accept param by reference it would have to be:
```cpp
inline SResourcePack operator * (const SResourcePack& pack, float scale) {
  SResourcePack copyToBeReturned = pack;
  return copyToBeReturned *= scale;
}
```
Accepting by value means we already get a copy.

Reply:
I think it's quite elegant, it seems to be a common idiom (e.g. check the examples in https://en.cppreference.com/w/cpp/language/operators, the page also says it helps optimize `x * y * z` chained calls somehow?) and ultimately it's boilerplate that nobody should really need to read or work with, but if you think it's too confusing I can make it more explicit.

Reply:
Oh yes, I misread, though I'm more used to seeing `operator *=(float scale)` in such cases.

Reply:
I think I wanted to add the symmetrical operator where the float is on the left (i.e. so you can do `2 * resources` instead of only `resources * 2`) which must be implemented as standalone like this. I forgot why ultimately I didn't but maybe it's a good idea to help future devs use the class.